packages/parser/lib/css_lexer.re

```diff
@@ -928,3 +933,51 @@ let from_string = string => {
 
   read([]);
 };
+
+let tokenize = input => {
+  let buffer = Sedlexing.Utf8.from_string(input);
+  let rec from_string = acc => {
+    switch (get_next_tokens_with_location(buffer)) {
+    | (Parser.EOF, _, _) => []
+    | token => [token, ...from_string(acc)]
+    };
+  };
+
+  try(Ok(from_string([]))) {
+  | exn => Error(Printexc.to_string(exn))
+  };
+};
+
+let render_token =
+  fun
+  | Parser.EOF => ""
+  | t => Tokens.token_to_debug(t);
+
+let position_to_string = pos =>
+  Printf.sprintf(
+    "[%d,%d+%d]",
+    pos.Lexing.pos_lnum,
+    pos.Lexing.pos_bol,
+    pos.Lexing.pos_cnum - pos.Lexing.pos_bol,
+  );
+
+let debug_token = ((token, loc_start, loc_end)) => {
```

Comment:
Instance of https://github.com/davesnx/styled-ppx/pull/444#discussion_r1522164658